platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/CopiedEncryptionService.java

```diff
@@ -0,0 +1,176 @@
+                return defaultPwDir;
+            }
+            if (!karafHomeDir.isDirectory()) {
+                System.out.println("Karaf home isn't directory: " + karafHome);
+                return defaultPwDir;
+            }
+            return karafHome.concat("/etc/certs");
+        }
+        System.out.println("Karaf home was null, defaulting to " + defaultPwDir);
+        return defaultPwDir;
+    }
+
+    public CopiedEncryptionService() {
+        System.out.println("KARAF HOME: " + System.getProperty("karaf.home"));
+        String passwordDirectory = getPasswordDirectory();
+
+        synchronized (CopiedEncryptionService.class) {
+            if (!new File(passwordDirectory.concat("/meta")).exists()) {
+                KeyczarTool.main(new String[] {"create", "--location=" + passwordDirectory,
+                        "--purpose=crypt", "--name=Password"});
+                KeyczarTool.main(new String[] {"addkey", "--location=" + passwordDirectory,
+                        "--status=primary"});
+            }
+            Crypter crypter = null;
+            try {
+                crypter = new Crypter(passwordDirectory);
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+            }
+
+            this.crypter = crypter;
+        }
+    }
+
+    /**
+     * Encrypts a plain text value using Keyczar.
+     *
+     * @param plainTextValue The value to encrypt.
+     */
+    public synchronized String encrypt(String plainTextValue) {
+        try {
+            return crypter.encrypt(plainTextValue);
+        } catch (Exception e) {
+            System.out.println("Key and encryption service failed to set up. Failed to encrypt. "
+                    + e.getMessage());
+            return plainTextValue;
+        }
+    }
+
+    /**
+     * Decrypts a plain text value using Keyczar
+     *
+     * @param encryptedValue The value to decrypt.
+     */
+    public synchronized String decrypt(String encryptedValue) {
+        try {
+            return crypter.decrypt(encryptedValue);
+        } catch (Exception e) {
+            System.out.println("Key and encryption service failed to set up. Failed to decrypt. "
+                    + e.getMessage());
+            return encryptedValue;
+        }
+    }
+    //
+    //    // @formatter:off
+//
+//    /**
+//     * @param wrappedEncryptedValue
+//     *            An wrapped encrypted password.
+//     *
+//     *            <pre>
+//     *   {@code
+//     *     One can encrypt passwords using the security:encrypt console command.
+//     *
+//     *     user@local>security:encrypt secret
+//     *     c+GitDfYAMTDRESXSDDsMw==
+//     *
+//     *     A wrapped encrypted password is wrapped in ENC() as follows: ENC(HsOcGt8seSKc34sRUYpakQ==)
+//     *   }
+//     * </pre>
+//     */
+//    // @formatter:on
+    //    public String decryptValue(String wrappedEncryptedValue) {
```

Comment:
@brjeter Do we need all this commented out logic?

Reply:
No, we don't.